src/compiler/crystal/codegen/codegen.cr

```diff
@@ -2066,20 +2069,20 @@ module Crystal
       struct_type = llvm_struct_type(type)
       if type.passed_by_value?
         type_ptr = alloca struct_type
+        memset type_ptr, int8(0), size_t(struct_type.size)
       else
         if type.is_a?(InstanceVarContainer) && !type.struct? &&
            type.all_instance_vars.each_value.any? &.type.has_inner_pointers?
-          type_ptr = malloc struct_type
+          type_ptr = calloc struct_type
         else
-          type_ptr = malloc_atomic struct_type
+          type_ptr = calloc_atomic struct_type
         end
       end
 
       pre_initialize_aggregate(type, struct_type, type_ptr)
     end
 
     def pre_initialize_aggregate(type, struct_type, ptr)
-      memset ptr, int8(0), size_t(struct_type.size)
```

Comment:
`.pre_initialize` must clear the memory because it is intended for use with uninitialized memory, regardless of where that memory came from. If the change in `#allocate_aggregate` is still needed then it should not call `#pre_initialize_aggregate`

Reply:
Then the naming `#pre_initialize_aggregate` isn't accurate anymore, because it now does less than the corresponding primitive. I suggest renaming it to something more descriptive

Reply:
But maybe this is also fine and the `.pre_initialize` primitive itself shouldn't clear the memory, so the caller can save every bit of redundant `memset`?